generator_test.go

```diff
@@ -66,147 +66,183 @@ func TestConcurrentAccess(t *testing.T) {
 }
 
 func TestGenerateResponseData(t *testing.T) {
-	var data interface{}
-	var err error
-	var generator DataGenerator
-
 	// basic reference
-	generator = DataGenerator{testSpec.Components.Schemas, &testFixtures}
-	data, err = generator.Generate(&GenerateParams{
-		Schema: &spec.Schema{Ref: "#/components/schemas/charge"},
-	})
-	assert.Nil(t, err)
-	assert.Equal(t,
-		testFixtures.Resources["charge"].(map[string]interface{})["id"],
-		data.(map[string]interface{})["id"])
-
-	// Makes sure that customer is *not* expanded
-	assert.Equal(t,
-		testFixtures.Resources["customer"].(map[string]interface{})["id"],
-		data.(map[string]interface{})["customer"])
+	{
```

Comment:
The diff in this test is a bit of a mess unfortunately — most of this didn't change, but I ended up wrapping each individual case in its own scope (i.e., `{ ... }`) just to help guarantee that we're not sharing tests between any state between them.